src/v/cloud_storage/remote_partition.cc

```diff
@@ -415,18 +415,27 @@ class partition_record_batch_reader_impl final
                         co_await set_end_of_stream();
                         throw std::system_error(result.error());
                     }
-                    data_t d = std::move(result.value());
-                    for (const auto& batch : d) {
+
+                    remote_segment_batch_reader::data_t d = std::move(
+                      result.value());
+                    data_t slice;
+                    slice.reserve(d.size());
```

Comment:
why do we need to copy? this code is part of the hot path, I'd like to avoid making unnecessary copies

Reply:
The bulk of this PR revolves around switching from a circular buffer to a chunked fifo object in remote segment, as to avoid a oversized allocation that occurred in a CI failure. However, the lower level code in Redpanda still expects a circular buffer as a storage type from remote partition downward, and so here is where we move the contents of the chunked fifo back into a circular buffer. 

As stated in the PR description, I agree that this seems like a band aid fix, and wanted to gauge what others thought about it as a means to avoid potential oversized allocations at this level.